Classes/EventStore/Storage/Doctrine/DoctrineEventStorage.php

```diff
@@ -117,18 +117,18 @@ public function load(StreamName $streamName, int $minimumSequenceNumber = 0): Ev
 
     /**
      * @inheritdoc
-     * @throws DBALException | ConcurrencyException
+     * @throws DBALException | ConcurrencyException | InvalidArgumentException | RuntimeException
      */
     public function commit(StreamName $streamName, WritableEvents $events, int $expectedVersion = ExpectedVersion::ANY): void
     {
         if ($streamName->isVirtualStream()) {
             throw new \InvalidArgumentException(sprintf('Can\'t commit to virtual stream "%s"', $streamName), 1540632984);
         }
         $this->reconnectDatabaseConnection();
-        $this->connection->beginTransaction();
-        if ($this->connection->getTransactionNestingLevel() > 1) {
+        if ($this->connection->getTransactionNestingLevel() > 0) {
             throw new \RuntimeException('A transaction is active already, can\'t commit events!', 1547829131);
         }
+        $this->connection->beginTransaction();
```

Comment:
Do I get it right that you merely moved the nesting level check _before_ the `beginTransaction` call? That makes sense of course

Reply:
Correct :)